src/Language/PureScript/Make/Actions.hs

```diff
@@ -236,12 +252,30 @@ buildMakeActions outputDir filePathMap foreigns usePrefix =
         Just path
           | not $ requiresForeign m ->
               tell $ errorMessage' (CF.moduleSourceSpan m) $ UnnecessaryFFIModule mn path
-          | otherwise ->
-              checkForeignDecls m path
+          | otherwise -> do
+              (foreignModuleType, foreignIdents) <- checkForeignDecls m path
+              case foreignModuleType of
+                ESModule -> copyFile path (outputFilename mn "foreign.js")
+                CJSModule -> do
+                  tell $ errorMessage' (CF.moduleSourceSpan m) $ DeprecatedFFICommonJSModule mn path
+                  copyFile path (outputFilename mn "foreign.cjs")
+                  writeESForeignModuleWrapper mn foreignIdents
+
         Nothing | requiresForeign m -> throwError . errorMessage' (CF.moduleSourceSpan m) $ MissingFFIModule mn
                 | otherwise -> return ()
-      for_ (mn `M.lookup` foreigns) $ \path ->
-        copyFile path (outputFilename mn "foreign.js")
+
+  writeESForeignModuleWrapper :: ModuleName -> S.Set Ident -> Make ()
+  writeESForeignModuleWrapper mn idents =
+    writeTextFile (outputFilename mn "foreign.js") . toByteString . renderJS $
+      JSAST.JSAstModule
+        [ JSAST.JSModuleExportDeclaration JSAST.JSNoAnnot
+            (JSAST.JSExportFrom
+              (JSAST.JSExportClause JSAST.JSAnnotSpace
+                (JSAST.toCommaList $ JSAST.JSExportSpecifier . JSAST.JSIdentName JSAST.JSAnnotSpace . T.unpack . runIdent <$> S.toList idents)
+                JSAST.JSAnnotSpace)
+              (JSAST.JSFromClause JSAST.JSAnnotSpace JSAST.JSAnnotSpace "\"./foreign.cjs\"")
+              (JSAST.JSSemi JSAST.JSNoAnnot))
+        ] JSAST.JSNoAnnot
```

Comment:
I hoped to write an `export * from "./foreign.cjs"` declaration here but this isn’t supported by `language-javascript-0.7.0.0`.

Reply:
I think this is preferable even if `export *` becomes available—enumerating the exports probably makes static analysis of the output a little easier, for both machines and brains.